server/queue/queue-rabbitmq/src/test/java/org/apache/james/queue/rabbitmq/RabbitMQMailQueueTest.java

```diff
@@ -174,6 +179,78 @@ void browseShouldReturnCurrentlyEnqueuedMailFromAllSlices() throws Exception {
                 "5-1", "5-2", "5-3", "5-4", "5-5");
         }
 
+        @Test
+        void dequeueShouldDeleteBlobs(CassandraCluster cassandra) throws Exception {
+            String name1 = "myMail1";
+            Flux<MailQueue.MailQueueItem> dequeueFlux = Flux.from(getMailQueue().deQueue());
+            getMailQueue().enQueue(defaultMail()
+                .name(name1)
+                .build());
+
+            dequeueFlux.take(1)
+                .flatMap(mailQueueItem -> Mono.fromCallable(() -> {
+                    mailQueueItem.done(true);
+                    return mailQueueItem;
+                })).blockLast(Duration.ofSeconds(10));
+
+            assertThat(cassandra.getConf().execute(select().from(BlobTables.DefaultBucketBlobTable.TABLE_NAME)))
+                .isEmpty();
+        }
+
+        @Test
+        void clearShouldDeleteBlobs(CassandraCluster cassandra) throws Exception {
+            String name1 = "myMail1";
+            getMailQueue().enQueue(defaultMail()
+                .name(name1)
+                .build());
+
+            getManageableMailQueue().clear();
+
+            assertThat(cassandra.getConf().execute(select().from(BlobTables.DefaultBucketBlobTable.TABLE_NAME)))
+                .isEmpty();
+        }
+
+        @Test
+        void removeByNameShouldDeleteBlobs(CassandraCluster cassandra) throws Exception {
+            String name1 = "myMail1";
+            getMailQueue().enQueue(defaultMail()
+                .name(name1)
+                .build());
+
+            getManageableMailQueue().remove(ManageableMailQueue.Type.Name, name1);
+
+            assertThat(cassandra.getConf().execute(select().from(BlobTables.DefaultBucketBlobTable.TABLE_NAME)))
+                .isEmpty();
+        }
+
+        @Test
+        void removeByRecipientShouldDeleteBlobs(CassandraCluster cassandra) throws Exception {
+            String name1 = "myMail1";
+            getMailQueue().enQueue(defaultMailNoRecipient()
+                .name(name1)
+                .recipient(RECIPIENT1)
+                .build());
+
+            getManageableMailQueue().remove(ManageableMailQueue.Type.Recipient, RECIPIENT1.asString());
+
+            assertThat(cassandra.getConf().execute(select().from(BlobTables.DefaultBucketBlobTable.TABLE_NAME)))
+                .isEmpty();
+        }
+
+        @Test
+        void removeBySenderShouldDeleteBlobs(CassandraCluster cassandra) throws Exception {
+            String name1 = "myMail1";
+            getMailQueue().enQueue(defaultMail()
+                .name(name1)
+                .sender(SENDER)
+                .build());
+
+            getManageableMailQueue().remove(ManageableMailQueue.Type.Sender, SENDER.asString());
+
+            assertThat(cassandra.getConf().execute(select().from(BlobTables.DefaultBucketBlobTable.TABLE_NAME)))
+                .isEmpty();
```

Comment:
do you have a test checking that if there are 2 mails in the queue, and we remove one, the matching blobs are removed but not the remaining ones?